### ISSUE TITLE
Trunc store for i8 not supported

### ISSUE THREAD

**Issue by ghost** [llvm-dcpu16/llvm-dcpu16#117]
Following code breaks:

```
char func(char s) {
    return s;
}
```

Error:

```
Do not know how to expand this store!
```

Breaking LLVM code:

```
ch = store 0x39fb750, 0x3a20800, 0x3a20a00, 0x3a20c00<ST1[%s.addr], trunc to i8>
```

**Comment by ghost** [comment]
Okay, my first try doesn't work. I tried to handle this truncstore with a peephole pattern:

```
def TRUNC16i8 : I16rm<0x0,
                    (outs GR16:$dst), (ins GR16:$src),
                    "AND\t{$dst, 0xFF} ; The Notch order",
                    []>;

def : Pat<(truncstorei8 (i16 GR16:$src), addr:$dst), (MOV16mr addr:$dst, (TRUNC16i8 GR16:$src))>;
```

(This looks oddly like a zero extend...)

Any ideas how to solve this problem gracefully?

**Comment by Blei** [comment]
fixed by #133
